scripts/release-crate.ps1

```diff
@@ -0,0 +1,464 @@
+            $description = $conventionalMatch.Groups[2].Value
+            $isConventional = $true
+        }
+
+        if ($isConventional -and $script:IgnoredTypes -contains $type) {
+            continue
+        }
+
+        if (-not [string]::IsNullOrEmpty($prBaseUrl)) {
+            $prMatch = $script:PrReferenceRegex.Match($description)
+            if ($prMatch.Success) {
+                $fullMatch = $prMatch.Groups[0].Value
+                $prNumber  = $prMatch.Groups[2].Value
+                $prLink    = " ([#$prNumber]($prBaseUrl/$prNumber))"
+                $description = $description.Substring(0, $description.Length - $fullMatch.Length) + $prLink
+            }
+        }
+
+        $groupKey = if ($script:TypeGroupMapping.ContainsKey($type)) { $script:TypeGroupMapping[$type] } else { $type }
+
+        if (-not $groupedCommits.Contains($groupKey)) {
+            $groupedCommits[$groupKey] = [System.Collections.ArrayList]::new()
+        }
+
+        [void]$groupedCommits[$groupKey].Add("  - $description")
+    }
+
+    $sortedKeys = Sort-KeysByPreferredOrder -allKeys $groupedCommits.Keys -preferredOrder $script:TypeOrder
+    $formattedLines = @()
+    foreach ($type in $sortedKeys) {
+        if ($groupedCommits[$type].Count -gt 0) {
+            $headerName = if ($script:HeaderNameMapping.ContainsKey($type)) { $script:HeaderNameMapping[$type] } else { $type.Substring(0, 1).ToUpper() + $type.Substring(1) }
+            $formattedLines += "- $headerName", "", $groupedCommits[$type], ""
+        }
+    }
+
+    if ($formattedLines.Count -gt 0 -and [string]::IsNullOrWhiteSpace($formattedLines[-1])) {
+        if ($formattedLines.Count -gt 1) {
+            $formattedLines = $formattedLines[0..($formattedLines.Count - 2)]
+        } else {
+            $formattedLines = @()
+        }
+    }
+
+    return $formattedLines
+}
+
+# --- SCRIPT FUNCTIONS ---
+
+function Update-CrateVersion {
+    param(
+        [string]$crateName,
+        [string]$version,
+        [string]$crateCargoToml,
+        [string]$rootCargoToml
+    )
+
+    $currentVersion = Get-CurrentVersion -cargoTomlPath $crateCargoToml
+
+    $newVersion = ""
+    if ([string]::IsNullOrEmpty($version)) {
+        $versionParts = $currentVersion.Split('.')
+        # Ensure versionParts has 3 elements (major, minor, patch)
+        while ($versionParts.Count -lt 3) {
+            $versionParts += '0'
+        }
+        $versionParts[1] = [int]$versionParts[1] + 1
+        $versionParts[2] = '0'
+        $newVersion = $versionParts -join '.'
+        Write-Host "✅ Auto-incrementing version from $currentVersion to $newVersion."
+    }
+    else {
+        $newVersion = $version
+        Write-Host "✅ Using specified version: $newVersion."
+    }
+
+    Write-Host "📝 Updating '$crateCargoToml'..."
+    (Get-Content $crateCargoToml -Raw) -replace '(?<=version\s*=\s*")[^"]+', $newVersion | Set-Content $crateCargoToml -NoNewline
```

Comment:
There is cargo set-version in cargo-edit crate - it will be more reliable than these regexes - it also bumps the versions in the dependencies table.

Reply:
No, I mean Cargo.tomls of crates that depend on this (you handle those manually on line 294).

Reply:
OK, let's see how it goes with this setup. The regexes are simple and so is the file format, so I feels robust.